src/main/java/eva/tasks/Deadline.java

```diff
@@ -0,0 +1,74 @@
+package eva.tasks;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import eva.exceptions.TaskException;
+
+/**
+ * Represents a task with a deadline. Extends the <code>Task</code> class.
+ * A <code>Deadline</code> object has a name, a status of whether it is done and a deadline.
+ */
+public class Deadline extends Task {
+    private LocalDate endTime;
+
+    /**
+     * Creates a new <code>Deadline</code> object with the specified name and deadline.
+     *
+     * @param name name of the deadline
+     * @param endTime deadline of the deadline
+     */
+    private Deadline(String name, LocalDate endTime) {
+        super(name);
+        this.endTime = endTime;
+    }
+
+    /**
+     * Creates a new <code>Deadline</code> object with the specified name, status and deadline.
+     *
+     * @param name name of the deadline
+     * @param isDone status of the deadline
+     * @param endTime deadline of the deadline
```

Comment:
Should these have punctuation at the end of the line?
```suggestion
     * @param name name of the deadline.
     * @param isDone status of the deadline.
     * @param endTime deadline of the deadline.
```

[Link to Relevant Line in Guide](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Punctuation%20behind%20each%20parameter%20description.)